Library/Formula/node.rb

```diff
@@ -55,42 +55,52 @@ def install
     system "./configure", *args
     system "make", "install"
 
-    resource("npm").stage libexec/"npm" if build.with? "npm"
-  end
+    if build.with? "npm"
+      resource("npm").stage buildpath/"npm_install"
 
-  def post_install
-    return if build.without? "npm"
+      # make sure npm can find node
+      ENV["PATH"] = "#{Formula["node"].prefix}/bin:#{ENV["PATH"]}"
 
-    (libexec/"npm").cd { system "make", "uninstall" }
-    Pathname.glob(HOMEBREW_PREFIX/"share/man/*") do |man|
-      next unless man.directory?
-      man.children.each do |file|
-        next unless file.symlink?
-        file.unlink if file.readlink.to_s.include? "/node_modules/npm/man/"
+      # set log level temporarily for npm's `make install`
+      ENV["NPM_CONFIG_LOGLEVEL"] = "verbose"
+
+      cd buildpath/"npm_install" do
+        system "./configure", "--prefix=#{libexec}/npm"
+        system "make", "install"
       end
+
+      rm_rf libexec/"npm/lib/node_modules/npm/share"
+
+      if build.with? "completion"
+        bash_completion.install \
+          buildpath/"npm_install/lib/utils/completion.sh" => "npm"
+      end
+
+      # Remove the now-redundant directory.
+      rm_rf buildpath/"npm_install"
     end
+  end
+
+  def post_install
+    return if build.without? "npm"
 
     node_modules = HOMEBREW_PREFIX/"lib/node_modules"
     node_modules.mkpath
-    cp_r libexec/"npm", node_modules
+    npm_exec = node_modules/"npm/bin/npm-cli.js"
+    # Kill npm but preserve all other modules across node updates/upgrades.
+    rm_rf node_modules/"npm"
```

Comment:
Was that a yay to exploring Tim's ideas of string dancing?

> why do manpages end up in man1 rather than libexec/"man1"?

For this install or for our previous ` node ` formula?